### PR TITLE
[WIP] Spark cross release

### PR DESCRIPTION
**Pull request by kianfar77**
Signed-off-by: kianfar77 <kiavash.kianfar@databricks.com>

## What changes are proposed in this pull request?
Spark cross release

## How is this patch tested?
- [ ] Unit tests
- [ ] Integration tests
- [ ] Manual tests

(Details)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -156,7 +160,7 @@ lazy val core = (project in file("core"))
   .settings(
     commonSettings,
     functionGenerationSettings,
-    name := "glow",
+    name := s"glow_spark_${sparkVersion.value}_scala",
```
Comment (karenfeng) [inline build.sbt:163]: This is a bit verbose - would this be `glow_spark_3.0.0`? Maybe we should shorten this to `glow_spark3`.

Comment (kianfar77) [reply build.sbt]: Changed to major version only.

**Comment by kianfar77** [comment]
Changed dependencies to settingKey to be able to use sparkVersion.value in defining them

File: build.sbt [hunk 2]
```diff
@@ -10,7 +10,18 @@ import sbt.nio.Keys._
...
 
-lazy val sparkVersion = sys.env.getOrElse("SPARK_VERSION", "3.0.0")
+lazy val spark2 = "2.4.5"
+lazy val spark3 = "3.0.0"
+
+lazy val sparkVersion = settingKey[String]("sparkVersion")
+ThisBuild / sparkVersion := sys.env.getOrElse("SPARK_VERSION", spark3)
+
+def majorVersion(version: String): String = {
+  StringUtils.ordinalIndexOf(version, ".", 1) match {
```
Comment (karenfeng) [inline build.sbt:20]: I think it'd be simpler to just have something along the lines of: `version.split('.').head`
